fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneMapQueryTest.java

```diff
@@ -0,0 +1,461 @@
+                                                    field("value"),
+                                                    field("key")))),
+                            value(null))));
+    private static final KeyExpression mapString2DoubleIndexExpression = field("stringToDoubleMap").nest(
+            function(LuceneFunctionNames.LUCENE_FIELD_NAME,
+                    concat(
+                            field("values", KeyExpression.FanType.FanOut).nest(
+                                    function(LuceneFunctionNames.LUCENE_FIELD_NAME,
+                                            concat(
+                                                    field("value"),
+                                                    field("key")))),
+                            value(null))));
+
+    private static final Index MAP_STRING_2_LONG_LUCENE_INDEX = new Index("MapField$string2long", concat(mapString2LongIndexExpression, field("doc_id")), LuceneIndexTypes.LUCENE);
+    private static final Index MAP_STRING_WRAPPER_TO_LONG_LUCENE_INDEX = new Index("MapField$stringWrapper2long", concat(mapStringWrapper2LongIndexExpression, field("doc_id")), LuceneIndexTypes.LUCENE);
+    private static final Index MAP_STRING_2_LONG_WRAPPER_LUCENE_INDEX = new Index("MapField$string2longWrapper", concat(mapString2LongWrapperIndexExpression, field("doc_id")), LuceneIndexTypes.LUCENE);
+    private static final Index MAP_STRING_2_INT_LUCENE_INDEX = new Index("MapField$string2int", concat(mapString2IntIndexExpression, field("doc_id")), LuceneIndexTypes.LUCENE);
+    private static final Index MAP_STRING_2_DOUBLE_LUCENE_INDEX = new Index("MapField$string2double", concat(mapString2DoubleIndexExpression, field("doc_id")), LuceneIndexTypes.LUCENE);
+
+    @BeforeAll
+    public static void setup() {
+        //set up the English Synonym Map so that we don't spend forever setting it up for every test, because this takes a long time
+        SynonymMapRegistryImpl.instance().getSynonymMap(EnglishSynonymMapConfig.ExpandedEnglishSynonymMapConfig.CONFIG_NAME);
+    }
+
+    final List<String> textSamples = Arrays.asList(
+            TextSamples.ROMEO_AND_JULIET_PROLOGUE,
+            TextSamples.AETHELRED,
+            TextSamples.ROMEO_AND_JULIET_PROLOGUE,
+            TextSamples.ANGSTROM,
+            TextSamples.AETHELRED,
+            TextSamples.FRENCH
+    );
+
+    private final List<TestRecordsTextProto.MapDocument> mapDocuments = IntStream.range(0, textSamples.size() / 2)
+            .mapToObj(i -> TestRecordsTextProto.MapDocument.newBuilder()
+                    .setDocId(i)
+                    .addEntry(TestRecordsTextProto.MapDocument.Entry.newBuilder().setKey("a").setValue(textSamples.get(i * 2)).build())
+                    .addEntry(TestRecordsTextProto.MapDocument.Entry.newBuilder().setKey("b").setValue(textSamples.get(i * 2 + 1)).build())
+                    .setStringToLongMap(TestRecordsTextProto.MapDocument.String2Long.newBuilder()
```

Comment:
All of these have exactly one entry in them, would it make sense to add another item to all of them.
Perhaps only if `i%2 == 0`?

Reply:
Done.